### ISSUE TITLE
consider default name when validating

### ISSUE THREAD

**Issue by caarlos0** [ldez/tagliatelle#6]
Hey! Thanks for this, amazing project!

I have a small issue which I think other people will have as well, for instance, consider this struct:

```go
type Foo struct {
  Foo string `yaml:",omitempty"`
  Bar string `yaml:"notBar"`
}
```

If I validate it using snake case, both fields should fail, as the first will be, by default `Foo` and the second one is in camel case.

Would be nice if either the default name is considered or if we could disallow not naming the field.

Cheers!

**Comment by ldez** [comment]
Hello,

I think I understand the use case, I will try to add a new option for this.

**Comment by ldez** [comment]
Fixed by 7b219253661560e60d284e3cddfb912dc49862fc
